### DISCUSSION TITLE
Why is my dbt run one block in the my Dagster run’s UI and gantt chart?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18115] [Q&A] [answered]
Hello - curious:

Why do each of my DBT files not appear as unique assets in a run, and instead come under run_dbt? Is there a way around this?

Secondly, I'm having a small issue with run_dbt being split up into two. Here is my scenario:
• I materialise an asset, then run dbt, then extract that table data and process it in a new asset, then run dbt again, then finally another asset follows to push it to elastic,
• On global linage view they are all hooked up correctly using dependencies.
• However if I select them all and materialise, they split into two run_dbts which aren't dependent. If run_dbt finishes before run_dbt2 then the final asset is incorrect as it does not have all the upstream data. But the final asset isn't waiting for run_dbt2

![image](https://github.com/dagster-io/dagster/assets/10715711/f047b24f-6165-46fc-a4c0-44714d532ed2)

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700230207826399?thread_ts=1700230207.826399&cid=C01U954MEER).*

**Comment by tacastillo** [comment, marked as answer]
Hi there!

The blocks on each chart you see in the run's page detail page is the step (otherwise known as Op) that your run is executing, and not necessarily what assets are being materialized. For what it's worth, many assets have a 1:1 relationship between the asset and the op that creates it, but the dbt integration is an exception and is a 1:many relationship of 1 op to many assets produced.

Dagster's dbt integration is optimized to create as few dbt runs as possible, and so it figures out what models to materialize in a dbt run/build. You can see what models each step is running from the logs. [Here's an example](https://github.com/dagster-io/dagster-open-platform/blob/cc90614c002a1e8ad692c822e89ca2d7be41aa8a/dagster_open_platform/assets/oss_analytics.py#L136) from one of our Dagster's runs.
<img width="979" alt="image" src="https://github.com/dagster-io/dagster/assets/10715711/dba40331-5555-47a2-a05d-3d5d67c749ee">


Compared to other orchestrators, others may make a task for each dbt model, which means that dbt model gets additional overhead time due to the task/executor getting spun up, which may be 1-3 seconds.

In your example, you have two runs. This is commonly found when Dagster needs to make different configurations across models, such as if you've partitioned a dbt model. A partitioned dbt run would have additional variables passed into it to tell the model what partition its running for.

Wrt to your second question, if you have incomplete data due to two runs, are the models across both marked dependent on each other? Dagster should resolve this by staggering, if so, but it might be a bug.
